central/jwt/jwt.go

```diff
@@ -52,6 +56,31 @@ func GetPrivateKeyBytes() ([]byte, error) {
 	return getBytesFromPem(privateKeyPathPEM)
 }
 
+type m2mValidator struct {
+	m2m.TokenExchangerSet
+	roxValidator tokens.Validator
+	issuerID     string
+}
+
+// Validate implements tokens.Validator.
+func (v *m2mValidator) Validate(ctx context.Context, token string) (*tokens.TokenInfo, error) {
+	iss, err := m2m.IssuerFromRawIDToken(token)
+	if err != nil {
+		return nil, err
+	}
+	if iss != v.issuerID {
+		exchanger, found := v.GetTokenExchanger(iss)
+		if !found {
+			return nil, errox.NoCredentials.CausedBy("no exchanger found for issuer " + iss)
+		}
+		token, err = exchanger.ExchangeToken(ctx, token)
```

Comment:
This is nasty name hiding. Please give tokens different names so it's clear what's happening there.

Reply:
Yes, thanks. I would have preferred `roxToken` but it's fine.